File: datasets/IR-Plag/original-java-rest-kotlin/case-07/non-plagiarized/09/t07.kt
package IR

import java.util.*

/*
* To change this license header, choose License Headers in Project Properties.
* To change this template file, choose Tools | Templates
* and open the template in the editor.
*/
/**
 *
 * @author 51BE04B11C5F2F5AD96A396632DC3F76
 */

object t07 {
    @JvmStatic
    fun main(args: Array<String>) {
        val sc = Scanner(System.`in`)
        val matriks = Array(4) { DoubleArray(4) }
        println("Enter a 4-by-4 matrix row by row: ")
        for (i in 0..3) {
            for (j in 0..3) {
                matriks[i][j] = sc.nextDouble()
            }
        }

        println(
            "Sum of the elements in the major diagonal is "
                    + sumMajorDiagonal(matriks)
        )
    }

    fun sumMajorDiagonal(m: Array<DoubleArray>): Double {
        var hasil = 0.0
        for (i in 0..3) {
            for (j in 0..3) {
                if (i == j) {
                    hasil += m[i][j]
                }
            }
        }
        return hasil
    }
}

File: datasets/IR-Plag/convertai-original-java-rest-typescript/case-07/non-plagiarized/09/t07.ts
import { Scanner } from 'readline-sync';

function sumMajorDiagonal(m: number[][]): number {
  let hasil = 0;
  for (let i = 0; i < 4; i++) {
    for (let j = 0; j < 4; j++) {
      if (i === j) {
        hasil += m[i][j];
      }
    }
  }
  return hasil;
}

function main(): void {
  const sc = new Scanner();
  const matriks: number[][] = new Array(4).fill(0).map(() => new Array(4).fill(0));
  console.log('Enter a 4-by-4 matrix row by row: ');
  for (let i = 0; i < 4; i++) {
    for (let j = 0; j < 4; j++) {
      matriks[i][j] = sc.nextDouble();
    }
  }

  console.log(`Sum of the elements in the major diagonal is ${sumMajorDiagonal(matriks)}`);
}

main();

